pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test: TestTenantLogicCCL_plpgsql_txn failed [3node-tenant tests are flaky with autocommit_before_ddl=true]

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogicCCL_plpgsql_txn [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18674670?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/18674670?buildTab=artifacts#/) on master @ [db0c794615911572c9c873e8c30ac9cd6d8b59f6](https://github.com/cockroachdb/cockroach/commits/db0c794615911572c9c873e8c30ac9cd6d8b59f6):


```
=== RUN   TestTenantLogicCCL_plpgsql_txn
    test_log_scope.go:165: test logs captured to: /artifacts/tmp/_tmp/647c645250a3b6a1f6a8541f942b2509/logTestTenantLogicCCL_plpgsql_txn1939806204
    test_log_scope.go:76: use -show-logs to present logs inline
[23:08:01] setting distsql_workmem='25576B';
[23:08:01] setting distsql_workmem='25576B';
    logic.go:4425: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/647c645250a3b6a1f6a8541f942b2509/logTestTenantLogicCCL_plpgsql_txn1939806204
--- FAIL: TestTenantLogicCCL_plpgsql_txn (10.96s)
=== RUN   TestTenantLogicCCL_plpgsql_txn/set_timestamp
    logic.go:3112: let $start_ts = 2025-01-30T23:08:04.023349Z
    logic.go:3112: let $ts_1 = 2025-01-30T23:08:03.023349Z
    logic.go:3112: let $ts_2 = 2025-01-30T23:08:02.023349Z
    logic.go:3112: let $ts_3 = 2025-01-30T23:08:01.023349Z
    logic.go:3112: let $ts_4 = 2025-01-30T23:08:00.023349Z
    logic.go:3112: let $ts_1 = 2025-01-30T23:08:04.194163Z
    logic.go:3112: let $ts_2 = 2025-01-30T23:08:04.227391Z
    logic.go:3112: let $ts_3 = 2025-01-30T23:08:04.255618Z
    logic.go:2273: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5931/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/plpgsql_txn:804: 
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=37aa2bcd key=/Tenant/10/Table/3/1/105/2/1 iso=Serializable pri=0.00288878 epo=0 ts=1738278484.284389934,0 min=1738278484.284389934,0 seq=16} lock=true stat=PENDING rts=1738278484.284389934,0 wto=false gul=1738278484.784389934,0
[23:08:10] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/5931/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/plpgsql_txn with config 3node-tenant: 73 tests, 1 failures
[23:08:11] --- total progress: 73 statements
--- total: 73 tests, 1 failures
    --- FAIL: TestTenantLogicCCL_plpgsql_txn/set_timestamp (6.42s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogicCCL_plpgsql_txn.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-47020

pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test.TestTenantLogicCCL_plpgsql_txn [failed](https://tanzanite.cluster.engflow.com/invocations/default/3788a372-18fe-437a-bb95-ae8d901147e8?testReportRun=1&testReportShard=44&testReportAttempt=1#targets-Ly9wa2cvY2NsL2xvZ2ljdGVzdGNjbC90ZXN0cy8zbm9kZS10ZW5hbnQ6M25vZGUtdGVuYW50X3Rlc3Q=) on master @ [5b931856d005214057f0703890ce3a77cdc21f30](https://github.com/cockroachdb/cockroach/commits/5b931856d005214057f0703890ce3a77cdc21f30):


```
=== RUN   TestTenantLogicCCL_plpgsql_txn
    test_log_scope.go:165: test logs captured to: outputs.zip/logTestTenantLogicCCL_plpgsql_txn3770882071
    test_log_scope.go:76: use -show-logs to present logs inline
[15:09:06] setting distsql_workmem='64870B';
[15:09:06] setting distsql_workmem='64870B';
    logic.go:4425: -- test log scope end --
test logs left over in: outputs.zip/logTestTenantLogicCCL_plpgsql_txn3770882071
--- FAIL: TestTenantLogicCCL_plpgsql_txn (10.93s)
=== RUN   TestTenantLogicCCL_plpgsql_txn/set_isolation
    logic.go:2273: 
         
        /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/plpgsql_txn:836: 
        expected success, but found
        (40001) restart transaction: TransactionRetryWithProtoRefreshError: TransactionAbortedError(ABORT_REASON_CLIENT_REJECT): "sql txn" meta={id=d52d5e38 key=/Tenant/10/Table/3/1/107/2/1 iso=Serializable pri=0.03859769 epo=0 ts=1738336148.912528580,0 min=1738336148.912528580,0 seq=24} lock=true stat=PENDING rts=1738336148.912528580,0 wto=false gul=1738336149.412528580,0
[15:09:14] --- done: /var/lib/engflow/worker/work/0/exec/bazel-out/k8-fastbuild/bin/pkg/ccl/logictestccl/tests/3node-tenant/3node-tenant_test_/3node-tenant_test.runfiles/com_github_cockroachdb_cockroach/pkg/ccl/logictestccl/testdata/logic_test/plpgsql_txn with config 3node-tenant: 75 tests, 1 failures
[15:09:15] --- total progress: 75 statements
--- total: 75 tests, 1 failures
    --- FAIL: TestTenantLogicCCL_plpgsql_txn/set_isolation (6.02s)
```

Parameters:
 - <code>attempt=1</code>
 - <code>run=1</code>
 - <code>shard=44</code>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestTenantLogicCCL_plpgsql_txn.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/spanconfig/spanconfigreconciler/reconciler.go
+++ b/pkg/spanconfig/spanconfigreconciler/reconciler.go
@@ -522,6 +522,16 @@ func (r *incrementalReconciler) reconcile(
 			) error {
 				var err error
 
+				// Using a fixed timestamp prevents this background job from contending
+				// with foreground schema change traffic. Schema changes modify system
+				// objects like system.descriptor, system.descriptor_id_seq, and
+				// system.span_count. The spanconfig reconciler needs to read these
+				// objects also. A fixed timestamp is a defensive measure to help
+				// avoid contention caused by this background job.
+				err = txn.KV().SetFixedTimestamp(ctx, checkpoint)
+				if err != nil {
+					return err
+				}
 				// TODO(irfansharif): Instead of these filter methods for missing
 				// tables and system targets that live on the Reconciler, we could
 				// move this to the SQLTranslator instead, now that the SQLTranslator

--- a/pkg/sql/logictest/logic.go
+++ b/pkg/sql/logictest/logic.go
@@ -1683,16 +1683,6 @@ func (t *logicTest) newCluster(
 		tenantID := serverutils.TestTenantID()
 		conn := t.cluster.SystemLayer(0).SQLConn(t.rootT)
 
-		// TODO(rafi): Remove this setting. We're adding it since the 3node-tenant
-		// config seem to be flaky with autcommit_before_ddl = true. Disable that
-		// setting for multitenant configs while the issue is being investigated.
-		if _, err := conn.Exec(
-			"ALTER TENANT [$1] SET CLUSTER SETTING sql.defaults.autocommit_before_ddl.enabled = false",
-			tenantID.ToUint64(),
-		); err != nil {
-			t.Fatal(err)
-		}
-
 		clusterSettings := toa.clusterSettings
 		if len(clusterSettings) > 0 {
 			// We reduce the closed timestamp duration on the host tenant so that the

